gemini/rag-engine/rag_engine_evaluation.ipynb

```diff
@@ -0,0 +1,983 @@
+        "    For each document in the BEIR corpus, we will create a new txt where:\n",
+        "      * doc_id will be the file name\n",
+        "      * doc_content will be the document text prepended by title (if any).\n",
+        "\n",
+        "    Args:\n",
+        "      corpus: BEIR corpus\n",
+        "      output_dir (str): Directory where the converted corpus will be saved\n",
+        "\n",
+        "    Returns:\n",
+        "      None (will write output to disk)\n",
+        "    \"\"\"\n",
+        "    # Create the output directory if it doesn't exist\n",
+        "    os.makedirs(output_dir, exist_ok=True)\n",
+        "\n",
+        "    file_count, subdir_count = 0, 0\n",
+        "    current_subdir = os.path.join(output_dir, f\"{subdir_count}\")\n",
+        "    os.makedirs(current_subdir, exist_ok=True)\n",
+        "\n",
+        "    # Convert each file in the corpus\n",
+        "    for doc_id, doc_content in corpus.items():\n",
+        "        # Combine title and text (if title exists)\n",
+        "        full_text = doc_content.get(\"title\", \"\")\n",
+        "        if full_text:\n",
+        "            full_text += \"\\n\\n\"\n",
+        "        full_text += doc_content[\"text\"]\n",
+        "\n",
+        "        # Create a new file for each file.\n",
+        "        file_path = os.path.join(current_subdir, f\"{doc_id}.txt\")\n",
+        "        with open(file_path, \"w\", encoding=\"utf-8\") as file:\n",
+        "            file.write(full_text)\n",
+        "\n",
+        "        file_count += 1\n",
+        "\n",
+        "        # Create a new subdirectory if the current one has reached the limit\n",
+        "        if file_count >= 10000:\n",
+        "            subdir_count += 1\n",
+        "            current_subdir = os.path.join(output_dir, f\"{subdir_count}\")\n",
+        "            os.makedirs(current_subdir, exist_ok=True)\n",
+        "            file_count = 0\n",
+        "\n",
+        "    print(f\"Conversion complete. {len(corpus)} files saved in {output_dir}\")\n",
+        "\n",
+        "\n",
+        "def count_files_in_gcs_bucket(bucket_path):\n",
+        "    \"\"\"\n",
+        "    Counts the number of files in a Google Cloud Storage bucket path,\n",
+        "    excluding directories.\n",
+        "\n",
+        "    Args:\n",
+        "      bucket_path: The path to the directory in the GCS bucket.\n",
+        "\n",
+        "    Returns:\n",
+        "      The number of files in the bucket path.\n",
+        "    \"\"\"\n",
+        "    # Add a trailing slash if it's not present\n",
+        "    if not bucket_path.endswith(\"/\"):\n",
+        "        bucket_path += \"/\"\n",
+        "\n",
+        "    process = subprocess.Popen(\n",
+        "        f\"gsutil ls {bucket_path}** | grep -v '/$' | wc -l\",\n",
+        "        shell=True,\n",
+        "        stdout=subprocess.PIPE,\n",
+        "        stderr=subprocess.PIPE,\n",
```

Comment:
Can you use the Google Cloud Storage python library instead of calling `subprocess`?

Reply:
Thanks for catching this! Done.